pkg/operator/controller/controller_router_deployment.go

```diff
@@ -244,16 +247,22 @@ func (r *reconciler) updateRouterDeployment(current, desired *appsv1.Deployment)
 // deploymentConfigChanged checks if current config matches the expected config
 // for the cluster ingress deployment and if not returns the updated config.
 func deploymentConfigChanged(current, expected *appsv1.Deployment) (bool, *appsv1.Deployment) {
-	// As per an offline conversation, this checks only the secret name
-	// for now but can be updated to a `reflect.DeepEqual` if needed.
-	if current.Spec.Template.Spec.Volumes[0].Secret.SecretName == expected.Spec.Template.Spec.Volumes[0].Secret.SecretName &&
+	if cmp.Equal(current.Spec.Template.Spec.Volumes, expected.Spec.Template.Spec.Volumes, cmpopts.EquateEmpty()) &&
```

Comment:
What would a narrowed version look like?

Comparing the desired route labels, e.g.

`metav1.LabelSelectorAsSelector(ci.Spec.RouteSelector)`

to the current route labels, e.g.

`labels.ParseSelector(env["ROUTE_LABELS"])`

?

Reply:
It could look something like that, for each volume and environment variable we wanted to check.

But should we ignore volumes and variables that are not on some check list? One of the requirements for an operator is to repair its operands.  https://github.com/openshift/cluster-ingress-operator/pull/94 attempted to do so by comparing the deployment's `.metadata.annotations`, `.metadata.labels`, and `.spec` and was reverted because it caused spurious reconciliations.  I believe checking volumes and environments as wholes should be sufficiently narrow to avoid the spurious reconciliations (https://github.com/openshift/cluster-ingress-operator/pull/124 and other recent clean-ups probably help too).  (I further believe it would be worthwhile to determine what in `.metadata.annotations`, `.metadata.labels`, and `.spec` was causing the spurious updates because the more we can check without causing spurious reconciles, the more we can repair or handle updates for.)

The alternative to checking volumes and environments as wholes is to hard-code volumes and a list of environment variables which matter enough that we need to ensure they have the right values, as opposed to volumes and environment variables we want to leave unmanaged.  I cannot imagine what the latter would be, and I cannot see the risk in simply checking all volumes and variables.

Reply:
> I cannot imagine what the latter would be, and I cannot see the risk in simply checking all volumes and variables.

Me either, but that's the risk we'll take: discovering in the wild what we're not foreseeing in this regard.

I'm okay to keep what you have here, let's just continue to be diligent with our diff/apply changes.